### PR TITLE
search frontend: show streaming loading, empty and done states

### PR DESCRIPTION
**Pull request by limitedmage**
Show states for streaming search when loading, done with no results, and done with some results.

### CODE REVIEW

**Review by keegancsmith — APPROVED** [review]

**Review by rvantonder — APPROVED** [review]
Is this PR the place where we can show "No results", or a follow up PR?

Context: https://sourcegraph.slack.com/archives/C0W2E592M/p1607052707325900?thread_ts=1607026378.319900&cid=C0W2E592M

File: client/web/src/search/results/streaming/StreamingSearchResults.tsx [hunk 1]
```diff
@@ -218,6 +220,24 @@ export const StreamingSearchResults: React.FunctionComponent<StreamingSearchResu
                     onShowMoreItems={onBottomHit}
                     items={results?.results.map(result => renderResult(result)).filter(isDefined) || []}
                 />
+
+                {(!results || !results.progress.done) && (
```
Comment (lguychard) [inline client/web/src/search/results/streaming/StreamingSearchResults.tsx:224]:
```suggestion
                {!results?.progress.done && (
```

**Review by lguychard — APPROVED** [review]

File: client/web/src/search/stream.tsx [hunk 2]
```diff
@@ -189,10 +189,7 @@ function toGQLFileMatchBase(fm: Omit<FileMatch, 'lineMatches' | 'type'>): GQL.IF
             oid: fm.version || '',
         },
     } as GQL.IGitBlob
-    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-    const repository: GQL.IRepository = {
-        name: fm.repository,
-    } as GQL.IRepository
+    const repository = toGQLRepositoryMatch({ type: 'repo', repository: fm.repository, branches: fm.branches })
```
Comment (felixfbecker) [inline client/web/src/search/stream.tsx:192]: nit unrelated to this PR: I had to expand the diff and read the context to understand what `fm` stands for here (`fileMatch`). We generally avoid abbreviations in TypeScript code.

Comment (limitedmage) [reply client/web/src/search/stream.tsx]: Thanks, updated this.

**Comment by limitedmage** [comment]
> Is this PR the place where we can show "No results", or maybe in a follow up PR?
> 
> Context: https://sourcegraph.slack.com/archives/C0W2E592M/p1607052707325900?thread_ts=1607026378.319900&cid=C0W2E592M

Yes! This will show loading indicator, 'no results' message, and count once search is done.

**Pull request merged**